lib/features/pokemon_details/pokemon_details_page.dart

```diff
@@ -19,6 +23,7 @@ class PokemonDetailsPage extends StatelessWidget {
       const StatsTabView(),
       const MovesTabView(),
     ];
+    final pokemonImageUrl = pokemon?.id.toString().toCustomUrlId ?? '';
```

Comment:
why do you still need this when you can just access the `url` property from your `pokemon` model instead of `id`? 🤔

Reply:
I was raising this issue because there is no url in the jsonstring so `pokemon.pokemon.url` actually returns null

Reply:
but you have something cached already right via dto? That's why I instructed you to create a `selectedPokemon` state and its respective `action` for you to utilize. Are we missing some context here? 🙂